### PR TITLE
changes in user crdential updations and readme

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Updated the user update flow in the authentication service to properly handle `username` and `password` changes.
- Revised the `README` in the `backend` to clarify the authentication flow and environment variable usage.

## Why

- Previously, changing the username or password caused the creation of a new user instead of updating the existing user credentials. This update addresses that issue and ensures the correct user is updated with the new credentials.

## How

- Fixed the user authentication logic to correctly update existing user credentials instead of creating a new user.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- There should be no other significant impact on existing features,  It alters how username and password updates are handled.

## Database Migrations

- No

## Env Config

- No

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -74,20 +74,30 @@ python manage.py runserver localhost:8000
...
-> **NOTE**: The username `admin` is reserved for Django admin, hence cannot be used
+To update the username or password after initial setup:
 
-To update the username or password after it's been set:
+1. Modify the credentials in `/backend/.env`
+   1. **DEFAULT_AUTH_USERNAME**=`your_new_username`
+   2. **DEFAULT_AUTH_PASSWORD**=`your_new_password`
+2. Save the `/backend/.env` file
+3. Restart the server to apply changes
+4. Login with the new credentials
```
[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/README.md:94]: @ritwik-g  Updated by adding a refernce in the Getting started Section .. Please check

File: backend/account_v2/authentication_service.py [hunk 2]
```diff
@@ -295,36 +303,108 @@ def get_organization_by_org_id(self, id: str) -> OrganizationData:
...
-            bool: True if the default user is successfully created and saved,
-                False otherwise.
+            User: The retrieved or created user
         """
-        if (
-            username != DefaultOrg.MOCK_USER
-            or password != DefaultOrg.MOCK_USER_PASSWORD
-        ):
-            return False
+        if not organization:
```
Comment (harini-venkataraman) [inline backend/account_v2/authentication_service.py:336]: @muhammad-ali-e [MINOR] Is this check needed here? Will there be a chance for `organization` to be None, when control reaches service layer?

File: README.md [hunk 3]
```diff
@@ -54,10 +54,13 @@ Next, either download a release or clone this repo and do the following:
 
 ✅ `./run-platform.sh`<br>
 ✅ Now visit [http://frontend.unstract.localhost](http://frontend.unstract.localhost) in your browser <br>
-✅ Use user name and password `unstract` to login
+✅ Use username and password `unstract` to login
+
 
 That's all there is to it!
 
+See [this guide](backend/README.md#authentication) for steps to change the default username and password.
```
Comment (chandrasekharan-zipstack) [inline README.md:62]:
```suggestion
Follow [these steps](backend/README.md#authentication) to change the default username and password.
```
NIT: For brevity

File: README.md [hunk 4]
```diff
@@ -54,10 +54,13 @@ Next, either download a release or clone this repo and do the following:
...
+
 See [user guide](https://docs.unstract.com/unstract/unstract_platform/user_guides/run_platform) for more details on managing the platform.  
 Another really quick way to experience Unstract is by signing up for our [hosted version](https://us-central.unstract.com/).
```
Comment (chandrasekharan-zipstack) [inline README.md:65]:
```suggestion
Another really quick way to experience Unstract is by signing up for our [hosted version](https://us-central.unstract.com/). It comes with a 14 day free trial!
```
NIT: Added this suggestion to nudge the OSS user slightly towards trying our hosted version

File: backend/account_v2/authentication_service.py (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply backend/account_v2/authentication_service.py:336]: @muhammad-ali-e if the above check is needed please change the mypy type of the argument in this function to indicate that its `Optional`

File: backend/account_v2/authentication_service.py [hunk 5]
```diff
@@ -295,36 +303,108 @@ def get_organization_by_org_id(self, id: str) -> OrganizationData:
...
+
+    def _promote_first_member_to_admin(self) -> Optional[OrganizationMember]:
+        """Promote the first organization member to admin role.
+
+        Returns:
+            Optional[OrganizationMember]: The promoted member if exists, None otherwise
+        """
+        members = OrganizationMemberService.get_members()
+        if not members:
+            logger.error("No organization member found")
```
Comment (chandrasekharan-zipstack) [inline backend/account_v2/authentication_service.py:387]: NIT: Will this error log be thrown for a fresh installation? If that's the case we should either mark it a warning or indicate in the log that its not an error somehow

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM for the most part, left minor comments

File: backend/README.md (hunk 1 above)
Comment (hari-kuriakose) [inline backend/README.md:94]: @muhammad-ali-e 

```suggestion
1. Modify the credentials in `/backend/.env`
    - **DEFAULT_AUTH_USERNAME**=`your_new_username`
    - **DEFAULT_AUTH_PASSWORD**=`your_new_password`
2. Save changes and restart `backend` service

Now you can login with the new credentials.
```

File: backend/account_v2/authentication_service.py (hunk 2 above)
Comment (muhammad-ali-e) [reply backend/account_v2/authentication_service.py:336]: @harini-venkataraman it needed for the firsttime to create organization.
@chandrasekharan-zipstack I added optional type

File: backend/account_v2/authentication_service.py (hunk 5 above)
Comment (muhammad-ali-e) [reply backend/account_v2/authentication_service.py]: updated as a warning

**Review by vishnuszipstack — APPROVED** [review]

**Pull request merged**